src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java

```diff
@@ -0,0 +1,161 @@
+                "FALSE");
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement()) {
+            int lines = stmt.executeUpdate(sql, 1);
+            System.out.println(lines);
+            if (lines != 0) {
+                try (ResultSet rs = stmt.getGeneratedKeys()) {
+                    if (rs.next()) {
+                        rs.last();
+                        flight.setId(rs.getLong("id"));
+                    }
+                } catch (SQLException e) {
+                    System.out.println("there");
+                    e.printStackTrace();
+
+                }
+                return flight;
+            }
+        } catch (SQLException k) {
+            k.printStackTrace();
+
+        }
+        flight = null;
+        return flight;
+    }
+
+    @Override
+    public Flight deleteFlight(Long id) {
+        Flight flight = getById(id);
+        flight.setDeleted(true);
+        String sql = String.format("UPDATE Flights SET isDeleted = TRUE WHERE id = '%d'", id);
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement()) {
+            int lines = stmt.executeUpdate(sql);
+            if (lines == 1) {
+                return flight;
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+
+        }
+        flight = null;
+        return flight;
+
+    }
+
+    @Override
+    public Flight updateFlight(Flight flight) {
+        if (getById(flight.getId()) == null)
+            return null;
+        String sql = String.format("UPDATE Flights SET initialPrice='%d',departureDate='%s'," +
+                        "arrivalDate='%s', plane_id='%d' WHERE id = %d",
+                flight.getInitialPrice(),
+                DateFormatter.format(flight.getDepartureDate()),
+                DateFormatter.format(flight.getArrivalDate()),
+                flight.getPlane().getId(),
+                flight.getId());
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement()) {
+            int lines = stmt.executeUpdate(sql);
+            if (lines == 1) {
+                return flight;
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+
+        }
+        flight = null;
+        return flight;
+    }
+
+    private Flight extractFlightFromRS(ResultSet rs) throws SQLException {
+        return Flight.builder()
+                .id(rs.getLong("id"))
+                .initialPrice(rs.getLong("initialPrice"))
+                .plane(extractPlaneFromRS(rs))
+                .departureDate(rs.getTimestamp("departureDate").toLocalDateTime())
+                .arrivalDate(rs.getTimestamp("arrivalDate").toLocalDateTime())
+                .isDeleted(rs.getBoolean("isDeleted"))
+                .build();
+    }
+
+    private Plane extractPlaneFromRS(ResultSet rs) throws SQLException {
```

Comment:
Has PlaneDAOImpl the same method ?

Reply:
yes, but I dont understand how to use it, so used PlaneRowMapper instead, is it ok?

Reply:
"let him go"